okcli/main.py

```diff
@@ -640,17 +644,31 @@ def refresh_completions(self, reset=False):
         if reset:
             with self._completer_lock:
                 self.completer.reset_completions()
+        status_message = 'Auto-completion refresh started in the background.'
+        if self.cache_completions:
+            maybe_cache = existing_completion_cache_file(self.sqlexecute.user, self.sqlexecute.host, self.config)
+            if maybe_cache is not None:
+              import pickle
+              status_message = "%s (using cache for now)" % status_message
+              with open(maybe_cache, 'rb') as compfile:
+                  new_completer = pickle.load(compfile)
+                  self._swap_completer_objects(new_completer)
         self.completion_refresher.refresh(
             self.sqlexecute, self._on_completions_refreshed,
             {'smart_completion': self.smart_completion,
              'supported_formats': self.formatter.supported_formats})
 
-        return [(None, None, None,
-                 'Auto-completion refresh started in the background.')]
+        return [(None, None, None, status_message)]
 
     def _on_completions_refreshed(self, new_completer):
         self._swap_completer_objects(new_completer)
 
+        if self.cache_completions:
+            import pickle
```

Comment:
And the  same here  encapsulating this in a method and a try-block to handle any error please.